File: backend/app/models/startup.py
from sqlalchemy import Column, Integer, String, Float, Boolean, Date
from app.db.base_class import Base

class Startup(Base):
    """Model representing a startup company."""

    # Define table name
    __tablename__ = "startups"

    # Primary key
    id = Column(Integer, primary_key=True, index=True)

    # Basic information
    name = Column(String, index=True)
    website = Column(String)
    industry = Column(String, index=True)
    sub_sector = Column(String)

    # Employee information
    employee_count = Column(Integer)
    local_employee_count = Column(Integer)
    headcount_growth = Column(Float)

    # Funding information
    total_funding = Column(Float)
    last_funding_date = Column(Date)
    funding_stage = Column(String)

    # Additional information
    is_hiring = Column(Boolean, default=False)
    last_updated = Column(Date)

    def __repr__(self):
        """Return a string representation of the Startup instance."""
        return f"<Startup(name={self.name}, industry={self.industry}, funding_stage={self.funding_stage})>"

File: frontend/src/types/startup.ts
// Define the Startup type representing a startup company
export type Startup = {
  // Unique identifier for the startup
  id: number;

  // Name of the startup
  name: string;

  // Website URL of the startup
  website: string;

  // Industry in which the startup operates
  industry: string;

  // Sub-sector or specific area within the industry
  sub_sector: string;

  // Total number of employees in the startup
  employee_count: number;

  // Number of employees located in the local area (e.g., Boston)
  local_employee_count: number;

  // Growth rate of the startup's headcount
  headcount_growth: number;

  // Total funding received by the startup (in USD)
  total_funding: number;

  // Date of the most recent funding round
  last_funding_date: Date;

  // Current funding stage of the startup (e.g., Seed, Series A, Series B)
  funding_stage: string;

  // Indicates whether the startup is currently hiring
  is_hiring: boolean;

  // Date when the startup's information was last updated
  last_updated: Date;
};